service/src/main/java/uk/gov/hmcts/reform/fpl/service/ChildrenService.java

```diff
@@ -30,4 +39,38 @@ public String getChildrenLabel(List<Element<Child>> children) {
     public void addPageShowToCaseDetails(CaseDetails caseDetails, List<Element<Child>> children) {
         caseDetails.getData().put("pageShow", children.size() <= 1 ? "No" : "Yes");
     }
+
+    public List<Element<Child>> updateFinalOrderIssued(List<Element<Child>> children,
+        String orderAppliesToAllChildren, ChildSelector childSelector) {
+
+        if (YES.getValue().equals(orderAppliesToAllChildren)) {
+            return children.stream()
+                .map(child -> updateFinalOrderIssuedOnChild(child, YES))
+                .collect(Collectors.toList());
+        } else {
+            List<Integer> selectedChildren = childSelector != null ? childSelector.getSelected() : new ArrayList<>();
```

Comment:
just a q, what is this child selector?

Reply:
defaultIfNull() static method would be nicer here too

Reply:
Child selector is set in the mid event. The Create Order UI allows you to select which child the order is issued for, this selection is stored by child selector.

Reply:
Unfortunately, defaultIfNull cannot be used here, as I am checking if childSelector is not null. It would work if I was checking childSelector.getSelected is null.